src/InstrumentationEngine/MethodInfo.cpp

```diff
@@ -246,14 +246,24 @@ HRESULT MicrosoftInstrumentationEngine::CMethodInfo::InitializeFullName()
     if (m_pDeclaringType)
     {
         CComBSTR declaringTypeName;
-        if (SUCCEEDED(m_pDeclaringType->GetName(&declaringTypeName)) && declaringTypeName.Length() > 0)
+        if (SUCCEEDED(m_pDeclaringType->GetName(&declaringTypeName)) &&
+            declaringTypeName != nullptr &&
+            declaringTypeName.Length() > 0)
         {
             nameBuilder << (LPWSTR)declaringTypeName << _T(".");
         }
     }
-    nameBuilder << (LPWSTR)m_bstrMethodName;
-    m_bstrMethodFullName = nameBuilder.str().c_str();
 
+    if (m_bstrMethodName != nullptr)
```

Comment:
Can we document/verify the expectations around m_bstrMethodName and m_bstrMethodFullName? It seems like m_bstrMethodName can be null but m_bstrMethodFullName is assigned the empty string. What do we want IMEthodInfo callers to get for these cases when they request the method name?

Reply:
I'm refactoring this so that m_bstrMethodName will never be null (ie. set to empty string). This essentially ensures all callers will get an empty string.